src/views/Mypage/pages/Mypage.tsx

```diff
@@ -0,0 +1,67 @@
+import { css } from '@emotion/react';
+import { useState } from 'react';
+
+import MenuBar from '@/components/MenuBar';
+
+import Favorite from '../components/Favorite';
+import Main from '../components/Main';
+import PersonalInfo from '../components/PersonalInfo';
+import TravelerType from '../components/TravelerType';
+
+export type currentTabType =
+  | 'main'
+  | 'personalInfo'
+  | 'favoritePlace'
+  | 'travelerType';
+
+const COMPONENT_LIST = [
+  { tab: 'personalInfo', component: PersonalInfo },
+  { tab: 'favoritePlace', component: Favorite },
+  { tab: 'travelerType', component: TravelerType },
+];
+
+function Mypage() {
+  const [currentTab, setCurrentTab] = useState<currentTabType>('main');
+
+  const handleSetCurrentTab = (clicked: currentTabType) => {
+    setCurrentTab(clicked);
+  };
+  return (
+    <>
+      {currentTab === 'main' && (
+        <div css={mypageContainer}>
+          <Main handleSetCurrentTab={handleSetCurrentTab} />
+          <footer css={footer}>
+            <MenuBar />
+          </footer>
+        </div>
+      )}
+
+      {COMPONENT_LIST.map(
+        ({ tab, component: Component }) =>
+          currentTab === tab && (
+            <div css={mypageContainer} key={tab}>
+              <Component handleSetCurrentTab={handleSetCurrentTab} />
+            </div>
+          ),
+      )}
```

Comment:
요고 항상 무조건 배열 돌면서 map돌려지는데 어차피 state 바뀔 때마다 리렌더링 되니까, 그냥 현재 tab에 따라 해당하는 컴포넌트 찾아서 바로 렌더링 시키는 함수를 이용하는 형태는 어떻게 생각하세욤 ?

Reply:
저두 그게 조금 더 좋아보여요!
그렇게 하면 지금 상수 중복되는 부분(`type`이랑 `COMPONENT_LIST` 부분)도 줄일 수 있을 것 같다는 생각이 드네요! 지금 문자 상수가 중복적으로 쓰이는 것 같아서 요 부분도 같이 정리하면 좋을 것 같아요 ㅎㅎ

Reply:
추가로, 지금 props drilling으로 `handleSetCurrentTab`랑 상위에서도 쓰는 상수들을 하위 컴포넌트들에서 쓰고 있는 것 같은데 제일 겉에 있는 `div` 박스만 내비두고 서현이 말처럼 조건부 렌더링으로 바로 하위 컴포넌트 불러오는 방법이 더 좋을 것 같다는 생각이 드네요!! 그러면 props drilling 안 해도 될 것 같은?! 혹시 이해가 되셨을련지....

```suggestion
const Contents = () => {
  if (step === Main) {
    return <Main />;
  } else if (step === personalInfo) {
    return <PersonalInfo />;
  } else if (step === favoritePlace) {
    return <FavoritePlace />;
  } else if (step === TravelerType) {
    return <TravelerType />;
  }
}

  return (
    <div css={mypageContainer}>
      {Content}
      {currentTab === 'main' && (
        <footer css={footer}>
          <MenuBar />
        </footer>
      )}
    </div>
  );
```

약간 이런 느낌으로? 아니면 `switch-case` 쓰셔도 될 것 같구요!

Reply:
모두 제안 고마워요 !! 수정해서 반영해두겠습니다❤️